taichi/codegen/codegen_opengl.cpp

```diff
@@ -68,24 +68,19 @@ struct CompiledKernel {
 struct CompiledProgram {
   std::vector<CompiledKernel> kernels;
   int arg_count;
-  int ext_arr_idx;
-  size_t ext_arr_size;
-  bool has_ext_arr{false};
+  std::map<int, size_t> ext_arr_map;
   size_t gtmp_size;
 
   CompiledProgram(Kernel *kernel, size_t gtmp_size) : gtmp_size(gtmp_size) {
-    has_ext_arr = false;
     arg_count = kernel->args.size();
     for (int i = 0; i < arg_count; i++) {
       if (kernel->args[i].is_nparray) {
-        if (has_ext_arr)
-          TI_ERROR(
-              "[glsl] external array argument is supported to at most one in "
-              "OpenGL for now");
-        ext_arr_idx = i;
-        ext_arr_size = kernel->args[i].size;
-        has_ext_arr = true;
+        ext_arr_map[i] = kernel->args[i].size;
       }
+      if (ext_arr_map.size() > 1)
+        TI_ERROR(
+            "[glsl] external array argument is supported to at most one in "
+            "OpenGL for now");
```

Comment:
I'm confused by this: isn't the goal of this PR to support more than one external arrays?

Reply:
Sorry! I forget that, how about to add a test for multiple extarr?

Reply:
Sounds good. 

I think currently we do have muti-extarr tests in `test_torch_io`. Those tests, however, will be skipped when the user does not have pytorch. It will be great to create a multi-external arr test suite without dependency on pytorch.